src/context/AppContext.tsx

```diff
@@ -0,0 +1,163 @@
+import {
+  createContext,
+  ReactNode,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { Todo } from '../types/todo';
+import useLocalStorage from '../hooks/useLocalStorage.hook';
+import { SortType } from '../enums/SortType';
+
+type AppContextContainerProps = {
+  todos: Todo[];
+  addNewTodo: (todoTitle: string) => void;
+  changeTitle: (id: number, value: string) => void;
+  clearCompleted: () => void;
+  ckickEsc: (id: number, pastTitle: string) => void;
+  editTodo: (id: number) => void;
+  makeTodosActive: () => void;
+  makeTodosCompleted: () => void;
+  removeTodo: (id: number) => void;
+  switchEdited: (id: number) => void;
+  todoCompleted: (id: number) => void;
+  sortType: SortType;
+  changeSortType: (option: SortType) => void;
+  inputRef: React.RefObject<HTMLInputElement>;
+};
+
+type Props = {
+  children: ReactNode;
+};
+
+const AppContextContainer = createContext({} as AppContextContainerProps);
+
+export const useAppContextContainer = () => {
+  return useContext(AppContextContainer);
+};
```

Comment:
I think I have writed comment about this today for your solution with fetched todos.